app/controllers/my/tracks_controller.rb

```diff
@@ -43,7 +43,23 @@ def show
     else
       core_exercises, side_exercises = normal_exercises.partition {|e|e.core?}
 
-      @core_exercises_and_solutions = core_exercises.map{|e|[e, mapped_solutions[e.id]]}
+      core_exercises = core_exercises.map do |e|
+        ExerciseWithSolution.new(e, mapped_solutions[e.id])
+      end
+
+      @core_exercises_and_solutions = core_exercises.
+        inject([[], [], []]) do |collection, exercise|
```

Comment:
I wonder if this would be nicer with `each_with_object` instead of inject so that we don't have to do L60.